homeassistant/components/refoss/__init__.py

```diff
@@ -0,0 +1,130 @@
+"""Refoss devices platform loader."""
+from __future__ import annotations
+
+from typing import Final, NamedTuple
+
+from refoss_ha.controller.device import BaseDevice
+from refoss_ha.device_manager import RefossDeviceListener, RefossDeviceManager
+from refoss_ha.socket_server import SocketServerProtocol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import dispatcher_send
+
+from .const import DOMAIN, LOGGER, REFOSS_DISCOVERY_NEW, REFOSS_HA_SIGNAL_UPDATE_ENTITY
+from .util import get_refoss_socket_server
+
+PLATFORMS: Final = [
+    Platform.SWITCH,
+]
+
+
+class HomeAssistantRefossData(NamedTuple):
+    """HomeAssistantRefossData."""
+
+    device_manager: RefossDeviceManager
+    device_listener: RefossDeviceListener
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Async setup  entry."""
+    if not entry.data.get(CONF_MAC):
+        LOGGER.debug(
+            (
+                "The config entry %s probably comes from a custom integration, please"
+                " remove it if you want to use core refoss integration"
+            ),
+            entry.title,
+        )
+        return False
+
+    hass.data.setdefault(DOMAIN, {})
+
+    device_ids: set[str] = set()
+    socketserver: SocketServerProtocol = await get_refoss_socket_server(hass)
+    device_manager = RefossDeviceManager(socket_server=socketserver)
+    await device_manager.async_start_broadcast_msg()
+    listener = DeviceListener(hass, device_manager, device_ids)
+    device_manager.add_device_listener(listener)
+
```

Comment:
`raise ConfigEntryNotReady` if the device is offline so it retries later

Reply:
ok